extra/resources/SystemHealth.in

```diff
@@ -203,44 +177,18 @@ SystemHealth_monitor() {
 }
 
 SystemHealth_validate() {
-
-    SystemHealth_check_tools
-    RC=$?
-
-    if [ $RC -ne 0 ]; then
-        return $RC
-    fi
-
-    return $OCF_SUCCESS
+    check_binary servicelog_notify
+    check_binary "$daemon_name"
+    check_binary "$OCF_RESKEY_program"
 }
 
-case $__OCF_ACTION in
-meta-data)      meta_data
-                exit $OCF_SUCCESS
-                ;;
-usage|help)     SystemHealth_usage
-                exit $OCF_SUCCESS
-                ;;
-esac
-
-SystemHealth_check_tools
-RC=$?
-
-if [ $RC -ne 0 ]; then
-        case "$__OCF_ACTION" in
-        stop)           exit $OCF_SUCCESS;;
-        *)              exit $RC;;
-        esac
-fi
-
 case "$__OCF_ACTION" in
-start)          SystemHealth_start;;
+start)          SystemHealth_validate; SystemHealth_start;;
 stop)           SystemHealth_stop;;
-monitor)        SystemHealth_monitor;;
-validate-all)   ;;
-*)              SystemHealth_usage
-                exit $OCF_ERR_UNIMPLEMENTED
-                ;;
+monitor)        SystemHealth_validate; SystemHealth_monitor;;
+validate-all)   SystemHealth_validate;;
+meta-data)      meta_data; exit $OCF_SUCCESS;;
+usage|help)     SystemHealth_usage; exit $OCF_SUCCESS;;
```

Comment:
I think we still want a catch-all for unimplemented actions

Reply:
Definitely. That was an accident while moving `meta-data` and `usage|help` downward.